integration-tests/src/tests/client/cold_storage.rs

```diff
@@ -384,31 +384,35 @@ fn test_initial_copy_to_cold(batch_size: usize) {
         last_hash = *block.hash();
     }
 
+    let keep_going = std::sync::Arc::new(std::sync::atomic::AtomicBool::new(true));
+
     copy_all_data_to_cold(
         (*store.cold_db().unwrap()).clone(),
         &env.clients[0].runtime_adapter.store(),
         batch_size,
+        keep_going,
     )
     .unwrap();
 
     for col in DBCol::iter() {
-        if col.is_cold() {
-            let num_checks = check_iter(
-                &env.clients[0].runtime_adapter.store(),
-                &store.get_store(Temperature::Cold),
-                col,
-                &vec![],
-            );
-            if col == DBCol::StateChangesForSplitStates || col == DBCol::StateHeaders {
-                continue;
-            }
-            // assert that this test actually checks something
-            assert!(num_checks > 0);
-            if batch_size == 0 {
-                assert_eq!(num_checks, test_get_store_initial_writes(col));
-            } else if batch_size == usize::MAX {
-                assert_eq!(1, test_get_store_initial_writes(col));
-            }
+        if !col.is_cold() {
+            continue;
+        }
+        let num_checks = check_iter(
+            &env.clients[0].runtime_adapter.store(),
+            &store.get_store(Temperature::Cold),
+            col,
+            &vec![],
+        );
+        if col == DBCol::StateChangesForSplitStates || col == DBCol::StateHeaders {
+            continue;
+        }
```

Comment:
Why is this if needed? Are those columns empty? If so can you add a comment?

Reply:
Those columns are empty in tests. Added comments